src/status_im/hardwallet/core.cljs

```diff
@@ -51,8 +80,10 @@
 (defn- proceed-to-pin-confirmation [fx]
   (assoc-in fx [:db :hardwallet :pin :enter-step] :confirmation))
 
-(defn- pin-match [fx]
-  (assoc-in fx [:db :hardwallet :pin :status] :validating))
+(defn- pin-match [{:keys [db]}]
+  {:db                   (assoc-in db [:hardwallet :pin :status] :validating)
+   :utils/dispatch-later [{:ms       3000
+                           :dispatch [:hardwallet.callback/on-pin-validated]}]})
```

Comment:
why the 3s delay?

Reply:
@goranjovic Removed this code as we don't allow set PIN for now

Reply:
@goranjovic i've addressed your and other guys comments, please take a look.